test/expected/index.out

```diff
@@ -0,0 +1,461 @@
+Child tables: _timescaledb_internal._hyper_4_3_chunk
+
+\d+ _timescaledb_internal._hyper*_chunk
+                    Table "_timescaledb_internal._hyper_4_3_chunk"
+ Column |           Type           | Modifiers | Storage | Stats target | Description 
+--------+--------------------------+-----------+---------+--------------+-------------
+ time   | timestamp with time zone |           | plain   |              | 
+ device | integer                  |           | plain   |              | 
+ temp   | double precision         |           | plain   |              | 
+Indexes:
+    "3_1_index_test_time_device_key" UNIQUE CONSTRAINT, btree ("time", device)
+    "_hyper_4_3_chunk_device_time_idx" btree (device, "time" DESC)
+    "_hyper_4_3_chunk_time_idx" btree ("time"), tablespace "tablespace2"
+    "_hyper_4_3_chunk_time_temp_idx" btree ("time", temp)
+Check constraints:
+    "constraint_3" CHECK ("time" >= 'Sat Dec 24 16:00:00 2016 PST'::timestamp with time zone AND "time" < 'Mon Jan 23 16:00:00 2017 PST'::timestamp with time zone)
+    "constraint_4" CHECK (_timescaledb_internal.get_partition_for_key(device::text) >= 1073741823 AND _timescaledb_internal.get_partition_for_key(device::text) < 2147483647)
+Inherits: index_test
+
+-- Constraint indexes NOT added to chunk_index table (maybe they
+-- should be for consistency?)
+SELECT * FROM _timescaledb_catalog.chunk_index;
+ chunk_id |            index_name            | hypertable_id |   hypertable_index_name    
+----------+----------------------------------+---------------+----------------------------
+        3 | _hyper_4_3_chunk_device_time_idx |             4 | index_test_device_time_idx
+        3 | _hyper_4_3_chunk_time_temp_idx   |             4 | index_test_time_temp_idx
+        3 | _hyper_4_3_chunk_time_idx        |             4 | index_test_time_idx
+(3 rows)
+
+DROP TABLE index_test;
+-- Metadata removed
+SELECT * FROM _timescaledb_catalog.chunk_index;
+ chunk_id | index_name | hypertable_id | hypertable_index_name 
+----------+------------+---------------+-----------------------
+(0 rows)
+
+-- Create table in a tablespace
+CREATE TABLE index_test(time timestamptz, temp float) TABLESPACE tablespace1;
+SELECT create_hypertable('index_test', 'time', create_default_indexes => false);
+ create_hypertable 
+-------------------
+ 
+(1 row)
+
+CREATE INDEX index_test_time_idx ON index_test (time);
+INSERT INTO index_test VALUES ('2017-01-20T09:00:01', 17.5);
+-- Index should be in table's tablespace
+\d+ index_test
+                              Table "public.index_test"
+ Column |           Type           | Modifiers | Storage | Stats target | Description 
+--------+--------------------------+-----------+---------+--------------+-------------
+ time   | timestamp with time zone |           | plain   |              | 
+ temp   | double precision         |           | plain   |              | 
+Indexes:
+    "index_test_time_idx" btree ("time")
```

Comment:
Hmm the index here does not appear in tablespace1. Shouldn't we be consistent with what postgres does?

Reply:
This index isn't really used so not sure? But I guess we could use the main index's tablespace and recurse to chunks instead of the hypertable's? The only downside is that you have to create all indexes in the "correct" tablespace. It gives you more control but less convenience. Although I see the correctness standpoint.